specification/purview/resource-manager/Microsoft.Purview/preview/2024-04-01-preview/purview.json

```diff
@@ -1631,6 +1630,64 @@
         }
       }
     },
+    "AccountMergeInfo": {
+      "description": "The public Account Merge Info model.",
+      "type": "object",
+      "properties": {
+        "accountLocation": {
+          "description": "The account location of the *other* account in the merge operation.",
+          "type": "string",
+          "readOnly": true
+        },
+        "accountName": {
+          "description": "The account name of the *other* account in the merge operation.",
+          "type": "string",
+          "readOnly": true
+        },
+        "accountResourceGroupName": {
+          "description": "The resource group name of the *other* account in the merge operation.",
+          "type": "string",
+          "readOnly": true
+        },
+        "accountSubscriptionId": {
+          "description": "The subscription id of the *other* account in the merge operation.",
+          "type": "string",
+          "readOnly": true
+        },
+        "deprovisioned": {
+          "description": "The deprovisioned status of the account.\r\nOnly applicable for the secondary account.",
+          "type": "boolean",
```

Comment:
Did you see the warning about how booleans are not recommended because they're not extensible? They also tend to lead to you having to combine many booleans, in order to express more states, when potentially one enum could have done the job.

Reply:
Yes, I did see the warning. For this case, a boolean should work as it is an irreversible state and only intended to be true or false. It is also scoped inside of the containing object as that is the only scenario it is intended to be set in and is for a specific use case and not a generalized use case.

Reply:
I get that you're thinking of deprovisioning as a one-time irreversible thing, but is it really something that happens independently and orthogonally of all the other state transitions your resource can go through?

Reply:
Yes, I totally understand your concerns, but in this specific case, it is intended to be one-time and irreversible. This field is merely an indicator of the account state and is intended to be transitional as well. Through this API, we will use this field to display some UX messages to a user. It is similar to a deleted state and is a one-way operation.